WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsViewModel.kt

```diff
@@ -135,8 +167,66 @@ class StatsViewModel
         }
     }
 
+    private fun updateDateSelector(statsSection: StatsSection = statsSectionManager.getSelectedSection()) {
+        val shouldShowDateSelection = statsSection != INSIGHTS
+
+        val updatedDate = getDateLabelForSection(statsSection)
+        val currentState = showDateSelector.value
+        val statsGranularity = statsSection.toStatsGranularity()
+        if ((!shouldShowDateSelection && currentState?.isVisible != false) || statsGranularity == null) {
```

Comment:
Perhaps it would be possible to simplify the update logic if we created a subclass of `LiveData` that would only get triggered if the new value is different. We could then just fire and forget without having to check for the current value.

Reply:
this is a good idea. It would be very useful to have this all over the place. What do you think about doing it in another PR? I think there is rarely a use case where we want to reemit the same item as before.

Reply:
That makes sense 👍.

Reply:
I've actually created a a Distinct LiveData in this PR but in this case it doesn't make sense because we only emit the section that has changed (for example `week`) every time we change the date - if we have a distinct call here, it stops working.